src/main/kotlin/snyk/container/annotator/ContainerYamlAnnotator.kt

```diff
@@ -0,0 +1,92 @@
+package snyk.container.annotator
+
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.ExternalAnnotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiFile
+import io.snyk.plugin.getSnykToolWindowPanel
+import org.jetbrains.annotations.TestOnly
+import snyk.container.ContainerIssuesForImage
+
+class ContainerYamlAnnotator : ExternalAnnotator<PsiFile, Unit>() {
+    private val logger = logger<ContainerYamlAnnotator>()
+
+    // overrides needed for the Annotator to invoke apply(). We don't do anything here
+    override fun collectInformation(file: PsiFile): PsiFile? = file
+    override fun doAnnotate(collectedInfo: PsiFile?): Unit = Unit
+
+    fun getContainerIssuesForImages(psiFile: PsiFile): List<ContainerIssuesForImage> {
+        logger.debug("Calling doAnnotate on ${psiFile.name}")
+
+        val containerResult = getSnykToolWindowPanel(psiFile.project)?.currentContainerResult
+
+        ProgressManager.checkCanceled()
+
+        return containerResult?.allCliIssues
+            ?.filter { forImage -> forImage.workloadImages.find { it.psiFile == psiFile } != null }
+            ?: emptyList()
+    }
+
+    override fun apply(psiFile: PsiFile, annotationResult: Unit, holder: AnnotationHolder) {
+        val issuesForImages = getContainerIssuesForImages(psiFile)
+
+        if (issuesForImages.isEmpty()) return
+
+        issuesForImages.forEach { forImage ->
+            if (forImage.ignored || forImage.obsolete) return
+
+            val severity = severity(forImage)
+
+            val workloadImage = forImage.workloadImages.first { it.psiFile == psiFile }
+            holder.newAnnotation(severity, annotationMessage(forImage))
+                .range(textRange(psiFile, workloadImage.lineNumber, forImage.imageName))
+                .create()
+        }
+    }
+
+    @TestOnly
+    fun severity(forImage: ContainerIssuesForImage): HighlightSeverity {
+        val severities = forImage.vulnerabilities.groupBy { it.severity }
+        return when {
```

Comment:
minor: some kind of comparator could be usefull here (also could be reused for other products)

Reply:
Let's write that comparator once we need it the second time!